src/commands/loader.ts

```diff
@@ -1,87 +0,0 @@
-import { Command } from 'commander';
-import fs from 'fs';
-import path from 'path';
-import { OrbitLogger } from '../utils/orbitLogger.helper';
-import { getScriptFileExtension } from '../utils/files.helper';
-import { ActionFunction } from 'src/actions/actionTypes';
-
-/**
- * Dynamically loads and registers command modules to the provided Commander program.
- * It scans the commands directory for files ending with `.command.ts` or `.command.js`,
- * depending on the environment (development or production), and then requires these files
- * to register their commands with the Commander program instance.
- *
- * Error handling is implemented to catch and report issues related to loading command files.
- * 
- * @param {Command} program - The Commander program instance to which the commands will be added.
- */
-export async function loadCommands(program: Command) {
-  const logger = new OrbitLogger('[LOADER]');
-
-  const commandsDir = path.join(__dirname, '.');
-  const fileExtension = getScriptFileExtension();
-
-  logger.debug(`Loading command files with extension: ${fileExtension}`);
-  logger.debug(`Commands directory: ${commandsDir}`);
-
-  try {
-    const commandFiles = fs
```

Comment:
Hello Rayan,

It's a pleasure to see the work. Regarding dynamic import, I thought it was an excellent feature for the development experience on the project. It allows us to avoid rewriting a lot of code with each new command, manually passing all the options to the CLI action, and having numerous program.addCommand in the bootstrap. In the end, it encourages having a *.action.ts file to improve readability and consistency in the project.

But we can discuss it through a meet including @adyngom to see what everyone thinks and then proceed after that.